### DISCUSSION TITLE
Cordova Microsoft OpenSDK

### DISCUSSION THREAD

**Discussion by pcwSlide** [apache/cordova#483] [Q&A]
Work is not licensing the required Cordova Java versions - money.
Does Cordova work with Microsoft OpenSDK?

**Comment by breautek** [comment]
I'm not finding anything called OpenSDK so I assume you mean Microsoft's [build](https://www.microsoft.com/openjdk) of OpenJDK.

It's not something that is tested against but if it implements the Java specification properly I see no reason why it wouldn't work.

[Cordova's CI](https://github.com/apache/cordova-android/blob/master/.github/workflows/ci.yml#L41) mainly uses [AdoptJDK](https://adoptium.net/en-GB/), which is also built from OpenJDK.

The JDK shipped with Android Studio could be used as well.

See [CORDOVA_JAVA_HOME](https://cordova.apache.org/docs/en/12.x/guide/platforms/android/index.html#java-development-kit-jdk) for how you can tell cordova which JDK environment to use.

**Comment by pcwSlide** [comment]
Thanks.. Yes I was pleased when they implemented the CORDOVA_JAVA_HOME idea - I think we originally discussed that on another issue.

Anyway - I was going on the requirements for the Android Platform implementation:
 
If you are using cordova-android 10.0.0 or greater, install the [Java Development Kit (JDK) 11]
If you are using any version below cordova-android 10.0.0, install the [Java Development Kit (JDK) 8]

Are you saying that say Microsoft Open JDK 21 should work?
Phil

**Reply by breautek** [reply]
you'll probably need JDK 17 if they have that version available (for current latest release, which the docs are currently slightly out-dated).

It's worth noting that Android Studio ships with an OpenJDK 17 build and that's what they use, if you were to do native android development.

**Comment by pcwSlide** [comment]
Hi Norman,
Look like the default structure works in 21 - 17 also for basic.
Q: On the work pc's they have:
C:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot\bin\java.exe
C:\Program Files\Microsoft\jdk-21.0.2.13-hotspot\bin\java.exe
I can't change this.

I use: ANDROID_JAVA_HOME = C:\Program Files\Android\Android 
Studio\jbr\bin
When run requirements it still says Java JDK: installed 21.0.2.

Should it not use C:\Program Files\Android\Android Studio\jbr\bin?
Phil

**Reply by breautek** [reply]
> When run requirements it still says Java JDK: installed 21.0.2.
>
> Should it not use C:\Program Files\Android\Android Studio\jbr\bin?

Depends on your environment variables.

Cordova will check `CORDOVA_JAVA_HOME` first and if set (and if it appears valid) it will use that JDK.

Otherwise Cordova fallsback to the standard `JAVA_HOME`. Cordova does not make use of `ANDROID_JAVA_HOME` environment variable.

**Reply by pcwSlide** [reply]
Apologies - ignore.

I come back to this module once a year. So I forgot that the path should not include the "bin" folder - as is common elsewhere. 

Anyway - sorry..

Phil
